spatialprofilingtoolbox/cggnn/extract.py

```diff
@@ -0,0 +1,57 @@
+"""Extract information cg-gnn needs from SPT."""
+
+from typing import Any
+
+from pandas import DataFrame, concat, merge  # type: ignore
+from numpy import sort  # type: ignore
+
+from spatialprofilingtoolbox.db.feature_matrix_extractor import FeatureMatrixExtractor
+
+
+def _create_cell_df(
+    dfs_by_specimen: dict[str, DataFrame],
+    feature_names: dict[str, str],
+) -> DataFrame:
+    """Find simple and complex phenotypes, and locations and merge into a DataFrame."""
+    feature_ids_to_names = {ft_id: 'FT_' + ft_name for ft_id, ft_name in feature_names.items()}
+    for specimen, df_specimen in dfs_by_specimen.items():
+        df_specimen.rename(feature_ids_to_names, axis=1, inplace=True)
+        # TODO: Create complex phenotype columns
+        df_specimen.rename({'pixel x': 'center_x', 'pixel y': 'center_y'},
+                           axis=1, inplace=True)
+        df_specimen['specimen'] = specimen
+
+    # TODO: Reorder so that it's simple phenotype, specimen, complex phenotype, x, y
+    df = concat(dfs_by_specimen.values(), axis=0)
```

Comment:
We can reorder the columns here once we figure out how to separate the simple and complex phenotypes.

I don't think it actually matters in practice but this is the ordering originally defined in `cg-gnn`'s own SQL-based extraction method.